### PR TITLE
[FLINK-15785][travis][e2e] Rework E2E test activations

### PR DESCRIPTION
**Pull request by zentol**
This PR reworks the activation logic for java end-to-end tests.

Introduces a new framework for specifying category inclusions/exclusions in `flink-end-to-end-tests`, that is capable of automatically aggregating properties across profiles.
Inclusions are defined by setting the `e2e.include.<category-class>`, conversely exclusions by setting `e2e.exclude.<category-class>`. The value of the property is a boolean that controls whether the inclusion/exclusions is enabled or not.
A groovy script searches for all properties matching the naming scheme, extracts the category class, aggregates inclusions/exclusions and sets the appropriate properties.

The script also prints the final inclusions/exclusions; I'd recommend trying it out and specifying different profile sets to see how it behaves.

**Background**: Maven does not really support combining categories. The surefire-plugin only accepts values like `catA,catB`, instead of a list where individual items have their own XML element. This implies that in order to combine category definitions one would have to concatenate the current value with some other value, i.e., do `categories = categories + "hello"`. Maven doesn't support this however since it is a recursive definition of a property.

The following is a listing of every problem and how it was addressed:

1) **travis activations are spread out over multiple files (.travis.yml, travis_watchdoh.sh)**
They are now consolidated in .travis.yml. While this does introduce some redundancy it is still preferable because .travis.yml is the most visible travis file, and it is easier to mvoe the pre-commit activation around.
2) **setting multiple categories in .travis.yml is incredibly verbose**
Each category now has their dedicated profile in `flink-end-to-end-tests` that set the appropriate category. This is significantly more concise and provides an abstraction of the specific categories used. (In other words, we can modify things at will without breaking setups)
3) **hadoop tests are not excluded by default**
Now disabled by default.
4) **hadoop exclusions aren't setup (causing FLINK-15685)** 
Are now setup.
5) **automatic java11 exclusion is prone to not working as intended**
There is now a dedicated java11 profile in `flink-end-to-end-tests` to ensure the java11 exclusions is always applied.

### CODE REVIEW

File: flink-end-to-end-tests/pom.xml [hunk 1]
```diff
@@ -35,8 +35,19 @@ under the License.
...
+		<!-- default inclusions / exclusions -->
+		<e2e.include.org.apache.flink.tests.util.categories.TravisGroup1>false</e2e.include.org.apache.flink.tests.util.categories.TravisGroup1>
+		<e2e.include.org.apache.flink.tests.util.categories.TravisGroup2>false</e2e.include.org.apache.flink.tests.util.categories.TravisGroup2>
+		<e2e.include.org.apache.flink.tests.util.categories.TravisGroup3>false</e2e.include.org.apache.flink.tests.util.categories.TravisGroup3>
+		<e2e.include.org.apache.flink.tests.util.categories.TravisGroup4>false</e2e.include.org.apache.flink.tests.util.categories.TravisGroup4>
+		<e2e.include.org.apache.flink.tests.util.categories.TravisGroup5>false</e2e.include.org.apache.flink.tests.util.categories.TravisGroup5>
+		<e2e.include.org.apache.flink.tests.util.categories.TravisGroup6>false</e2e.include.org.apache.flink.tests.util.categories.TravisGroup6>
+		<e2e.exclude.org.apache.flink.tests.util.categories.Hadoop>true</e2e.exclude.org.apache.flink.tests.util.categories.Hadoop>
+		<e2e.exclude.org.apache.flink.testutils.junit.FailsOnJava11>true</e2e.exclude.org.apache.flink.testutils.junit.FailsOnJava11>
```
Comment (tillrohrmann) [inline flink-end-to-end-tests/pom.xml]: Should `<e2e.include.org.apache.flink.tests.util.categories.PreCommit>false</e2e.include.org.apache.flink.tests.util.categories.PreCommit>` be added here?

File: flink-end-to-end-tests/pom.xml [hunk 2]
```diff
@@ -90,6 +162,66 @@ under the License.
...
+
+									def excludes = categorySearcher('e2e.exclude.')
+									// allow overrides from the command-line
+									if (!project.properties.excludeE2E.isEmpty()) {
+										excludes.add(project.properties.excludeE2E)
+									}
+									project.properties.excludeE2E = excludes.join(',')
+
+									println "includes: " + project.properties.includeE2E
+									println "excludes: " + project.properties.excludeE2E
```
Comment (tillrohrmann) [inline flink-end-to-end-tests/pom.xml]: Do we want to keep these `printlns`?

File: flink-end-to-end-tests/pom.xml (hunk 1 above)
Comment (zentol) [reply flink-end-to-end-tests/pom.xml]: we can. technically it's not necessary, but it's a good idea to do so for clarity.

File: flink-end-to-end-tests/pom.xml (hunk 2 above)
Comment (zentol) [reply flink-end-to-end-tests/pom.xml]: I'm not sure. I like that you can tell at a glance whether something is off, which will be useful on Travis.

Comment (zentol) [reply flink-end-to-end-tests/pom.xml]: maybe keep them for a bit until the transition to azure is done and we are confident it works as it should.

File: flink-end-to-end-tests/pom.xml (hunk 1 above)
Comment (zentol) [reply flink-end-to-end-tests/pom.xml]: done.

**Pull request merged**